src/planner/plan/append_extend.cpp

```diff
@@ -197,12 +197,12 @@ void Planner::appendRecursiveExtendAsGDS(const std::shared_ptr<NodeExpression>&
     pathPropertyProbe->extendFromLeft = *boundNode == *rel->getLeftNode();
     pathPropertyProbe->pathNodeIDs = recursiveInfo->pathNodeIDsExpr;
     pathPropertyProbe->pathEdgeIDs = recursiveInfo->pathEdgeIDsExpr;
-    pathPropertyProbe->getSIPInfoUnsafe().position = SemiMaskPosition::PROHIBIT;
     pathPropertyProbe->computeFactorizedSchema();
     probePlan.setLastOperator(pathPropertyProbe);
     probePlan.setCost(plan.getCardinality());
-    auto cardinality = UINT32_MAX;
-    probePlan.setCardinality(cardinality);
+    auto extensionRate =
```

Comment:
Can you add a comment here that we're adding a penalty factor UINT32_MAX here to ensure that in almost all cases, the GDS pipeline is on the probe side? This is a heuristic rule we keep before we have a better way to figure out a good estimation on output cardinality of recursive joins, such as based on the max degree of the rel table in the direction.